File: apps/api/app/schemas/patient_chart.py
from pydantic import BaseModel

from app.schemas.fax import FaxOut
from app.schemas.patient_document import PatientDocumentOut
from app.schemas.schedule import AppointmentOut
from app.schemas.task import TaskOut


class PatientChartSummaryCounts(BaseModel):
    documents_total: int
    documents_needing_review: int
    open_tasks: int
    urgent_tasks: int
    recent_faxes: int
    upcoming_appointments: int


class PatientChartSummaryOut(BaseModel):
    patient_id: str
    checkout_readiness: str
    blockers: list[str]
    counts: PatientChartSummaryCounts
    documents: list[PatientDocumentOut]
    open_tasks: list[TaskOut]
    recent_faxes: list[FaxOut]
    upcoming_appointments: list[AppointmentOut]

File: packages/shared/src/types/patient-chart.ts
import type { Appointment } from './schedule';
import type { Fax } from './fax';
import type { PatientDocument } from './patient-document';
import type { Task } from './task';
import type { UUID } from './common';

export interface PatientChartSummaryCounts {
  documents_total: number;
  documents_needing_review: number;
  open_tasks: number;
  urgent_tasks: number;
  recent_faxes: number;
  upcoming_appointments: number;
  unsigned_encounters: number;
  medications_needing_review: number;
  labs_needing_review: number;
  care_plan_blockers: number;
}

export interface PatientChartSummary {
  patient_id: UUID;
  checkout_readiness: 'ready' | 'blocked';
  blockers: string[];
  counts: PatientChartSummaryCounts;
  documents: PatientDocument[];
  open_tasks: Task[];
  recent_faxes: Fax[];
  upcoming_appointments: Appointment[];
}
